### PR TITLE
[FX-5300] Migrate TagSelector to Tailwind

### PR DESCRIPTION
**Pull request by augustobmoura**
[FX-5300]

### Description

Migrating JSS styles to TailwindCSS

### How to test

- [Temploy](https://picasso.toptal.net/fx-5300-migrate-tagselector)
- Check storybook page for TagSelector on temploy and Happo screenshots

### Screenshots

No screenshots

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5300]: https://toptal-core.atlassian.net/browse/FX-5300?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/wet-impalas-heal.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso-tagselector': minor
```
Comment (TomasSlama) [inline .changeset/wet-impalas-heal.md:2]:
```suggestion
'@toptal/picasso-tagselector': major
```
this package now requires Tailwind setup

File: packages/base/Tagselector/package.json [hunk 2]
```diff
@@ -35,10 +35,11 @@
     "**/styles.js"
   ],
   "peerDependencies": {
-    "@material-ui/core": "4.12.4",
+    "@toptal/picasso-tailwind": "2.6.0",
```
Comment (TomasSlama) [inline packages/base/Tagselector/package.json:38]:
```suggestion
    "@toptal/picasso-tailwind": ">=2.6.0",
```

File: packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx [hunk 3]
```diff
@@ -55,23 +46,25 @@ export const TagSelectorInput = forwardRef<HTMLInputElement, InputProps>(
...
-
-    let usedEndAdornment = null
+    let usedEndAdornment: React.ReactNode
 
-    if (endAdornment) {
-      usedEndAdornment = React.cloneElement(endAdornment as ReactElement, {
-        className: classes.endAdornment,
+    if (endAdornment && React.isValidElement(endAdornment)) {
+      usedEndAdornment = React.cloneElement(endAdornment, {
+        className: 'absolute top-[calc(50%-0.5*1em)] right-[0.625em]',
```
Comment (TomasSlama) [inline packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx:53]:
```suggestion
        className: 'absolute top-[calc(50%-0.5em)] right-[0.625em]',
```

File: packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx [hunk 4]
```diff
@@ -55,23 +46,25 @@ export const TagSelectorInput = forwardRef<HTMLInputElement, InputProps>(
...
 
     return (
       <OutlinedInput
         inputRef={ref}
         style={style}
-        className={cx(classes.inputBase, {
-          [classes.withEndAdornment]: Boolean(endAdornment),
-        })}
+        className={cx(
+          `flex flex-wrap h-auto pb-1 pl-1 pt-1
```
Comment (TomasSlama) [inline packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx:62]:
```suggestion
          `flex flex-wrap h-auto py-1 pl-1
```

File: packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx [hunk 5]
```diff
@@ -55,23 +46,25 @@ export const TagSelectorInput = forwardRef<HTMLInputElement, InputProps>(
...
+          [&>input]:min-w-[3em] [&>input]:flex-grow [&>input]:w-0 [&>input]:h-6 [&>input]:pl-1 [&>input]:pr-0 [&>input]:mb-0`,
+          {
+            'pr-[calc(2*0.625em+1em)]': Boolean(endAdornment),
```
Comment (TomasSlama) [inline packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx:65]:
```suggestion
            'pr-[calc(2*1.625em)]': Boolean(endAdornment),
```

**Review by TomasSlama — APPROVED** [review]
I left few small comments, looks good!

**Comment by augustobmoura** [comment]
@toptal-anvil ping reviewers

File: packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx (hunk 5 above)
Comment (augustobmoura) [reply packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx:65]: Here we have a different operator precedence for multiplication, so the correct value would actually be `1.25em + 1em` or, after simplifying it further: `2.5em`. Funny that I just copied how the padding was being made before, but didn't notice that the value is always a constant that can be simplified to a single number

File: packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx [hunk 6]
```diff
@@ -55,23 +46,25 @@ export const TagSelectorInput = forwardRef<HTMLInputElement, InputProps>(
...
 
     return (
       <OutlinedInput
         inputRef={ref}
         style={style}
-        className={cx(classes.inputBase, {
-          [classes.withEndAdornment]: Boolean(endAdornment),
-        })}
+        className={cx(
+          `flex flex-wrap h-auto py-1 pl-1
```
Comment (AdrianContiu) [inline packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx:62]: Here is missing `cursor-pointer`.

File: packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx [hunk 7]
```diff
@@ -55,23 +46,25 @@ export const TagSelectorInput = forwardRef<HTMLInputElement, InputProps>(
...
-
-    let usedEndAdornment = null
+    let usedEndAdornment: React.ReactNode
 
-    if (endAdornment) {
-      usedEndAdornment = React.cloneElement(endAdornment as ReactElement, {
-        className: classes.endAdornment,
+    if (endAdornment && React.isValidElement(endAdornment)) {
+      usedEndAdornment = React.cloneElement(endAdornment, {
+        className: 'absolute top-[calc(50%-0.5em)] right-[0.625em]',
```
Comment (AdrianContiu) [inline packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx:53]: It seems that there was also a height of `h-[1em]`

**Review by AdrianContiu — APPROVED** [review]
LGTM! I only left a few minor comments, please check them before merging.

File: packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx (hunk 6 above)
Comment (augustobmoura) [reply packages/base/Tagselector/src/TagSelectorInput/TagSelectorInput.tsx:62]: Good catch!

**Pull request merged**